akka-sse/src/main/scala/de/heikoseeberger/akkasse/ServerSentEvent.scala

```diff
@@ -68,10 +101,14 @@ object ServerSentEvent {
  * Representation of a Server-Sent Event.
  * @param data data which may be empty or span multiple lines
  * @param eventType optional event type, must not contain \n or \r
+ * @param id event id, must not contain \n or \r
+ * @param retry the reconnection time in milliseconds.
  */
-final case class ServerSentEvent(data: String, eventType: Option[String] = None) {
+final case class ServerSentEvent(data: String, eventType: Option[String] = None, id: Option[String] = None, retry: Option[Int] = None) {
   import ServerSentEvent._
   require(eventType.forall(_.forall(c => c != '\n' && c != '\r')), "Event type must not contain \\n or \\r!")
+  require(id.forall(_.forall(c => c != '\n' && c != '\r')), "Id must not contain \\n or \\r!")
+  require(retry.forall(_ > 0L), "Retry must be a positive number!")
```

Comment:
Yip:
> If the field value consists of only characters in the range U+0030 DIGIT ZERO (0) to U+0039 DIGIT NINE (9), then interpret the field value as an integer in base ten, and set the event stream's reconnection time to that integer. Otherwise, ignore the field.